src/data/node_data.go

```diff
@@ -55,12 +57,22 @@ func (n *NodeData) UnmarshalJSON(data []byte) error {
 	aux := &struct {
 		Partitions []interface{} `json:"partitions"`
 		*Alias
+		DiskFreeSpace *big.Int `json:"disk_free" metric_name:"node.diskSpaceFreeInBytes" source_type:"gauge"`
```

Comment:
why adding this and not change it in the NodeData?

Reply:
If I changed it in the NodeData big values would be rendered with no issue and reported as well, since [gauges must fit in the range of Java double](https://docs.newrelic.com/docs/data-apis/understand-data/metric-data/metric-data-type/) it will be rechecked in the back-end.

This way, we can check if the value fits in int64 and discard it (by setting it to nil) if it does not (as it's done below)

Reply:
Thanks i get it. I'm a bit concerned that the sdk is not doing something like this automatically since it could affect any of the other values.